### PR TITLE
[CN-625] improve logging

### PR DESCRIPTION
**Pull request by kutluhanmetin**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,11 +3,14 @@ module github.com/hazelcast/platform-operator-agent
 go 1.19
 
 require (
+	github.com/go-logr/logr v1.2.2
+	github.com/go-logr/zapr v1.2.3
```
Comment (dzeromski-hazelcast) [inline go.mod:7]: Please use go.uber.org/zap and not `github.com/go-logr/logr` :)

https://github.com/uber-go/zap

File: init/restore/bucket_to_hostpath_cmd.go [hunk 2]
```diff
@@ -4,7 +4,7 @@ import (
 	"context"
 	"flag"
 	"fmt"
-	"log"
+	"github.com/go-logr/logr"
```
Comment (dzeromski-hazelcast) [inline init/restore/bucket_to_hostpath_cmd.go:7]: Import grouping: https://google.github.io/styleguide/go/decisions#import-grouping

File: init/restore/bucket_to_hostpath_cmd_test.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package restore
 
 import (
 	"context"
+	"github.com/hazelcast/platform-operator-agent/internal/logger"
```
Comment (dzeromski-hazelcast) [inline init/restore/bucket_to_hostpath_cmd_test.go:5]: https://google.github.io/styleguide/go/decisions#import-grouping

File: init/restore/bucket_to_pvc_cmd.go [hunk 4]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"flag"
 	"fmt"
+	"github.com/go-logr/logr"
```
Comment (dzeromski-hazelcast) [inline init/restore/bucket_to_pvc_cmd.go:7]: https://google.github.io/styleguide/go/decisions#import-grouping

File: init/restore/bucket_to_pvc_cmd_test.go [hunk 5]
```diff
@@ -2,6 +2,7 @@ package restore
 
 import (
 	"context"
+	"github.com/hazelcast/platform-operator-agent/internal/logger"
```
Comment (dzeromski-hazelcast) [inline init/restore/bucket_to_pvc_cmd_test.go:5]: https://google.github.io/styleguide/go/decisions#import-grouping

File: init/restore/local_in_hostpath_cmd.go [hunk 6]
```diff
@@ -4,8 +4,8 @@ import (
 	"context"
 	"flag"
 	"fmt"
+	"github.com/go-logr/logr"
```
Comment (dzeromski-hazelcast) [inline init/restore/local_in_hostpath_cmd.go:7]: https://google.github.io/styleguide/go/decisions#import-grouping

File: init/restore/local_in_hostpath_cmd_test.go [hunk 7]
```diff
@@ -1,6 +1,7 @@
 package restore
 
 import (
+	"github.com/hazelcast/platform-operator-agent/internal/logger"
```
Comment (dzeromski-hazelcast) [inline init/restore/local_in_hostpath_cmd_test.go:4]: https://google.github.io/styleguide/go/decisions#import-grouping

File: init/restore/local_in_pvc_cmd.go [hunk 8]
```diff
@@ -4,7 +4,7 @@ import (
 	"context"
 	"flag"
 	"fmt"
-	"log"
+	"github.com/go-logr/logr"
```
Comment (dzeromski-hazelcast) [inline init/restore/local_in_pvc_cmd.go:7]: https://google.github.io/styleguide/go/decisions#import-grouping

File: init/usercode/user_code_deployment.go [hunk 9]
```diff
@@ -3,8 +3,8 @@ package usercode
 import (
 	"context"
 	"flag"
+	"github.com/go-logr/logr"
```
Comment (dzeromski-hazelcast) [inline init/usercode/user_code_deployment.go:6]: https://google.github.io/styleguide/go/decisions#import-grouping

File: main.go [hunk 10]
```diff
@@ -3,26 +3,31 @@ package main
...
 	subcommands.Register(subcommands.FlagsCommand(), "")
 	subcommands.Register(subcommands.CommandsCommand(), "")
 
-	subcommands.Register(&usercode.Cmd{}, "")
-	subcommands.Register(&restore.BucketToHostpathCmd{}, "")
-	subcommands.Register(&restore.LocalInHostpathCmd{}, "")
-	subcommands.Register(&restore.LocalInPVCCmd{}, "")
-	subcommands.Register(&restore.BucketToPVCCmd{}, "")
-	subcommands.Register(&sidecar.Cmd{}, "")
+	subcommands.Register(&usercode.Cmd{Logger: log.WithName("user code deployment")}, "")
```
Comment (dzeromski-hazelcast) [inline main.go:25]: Could we use approach similar to what we do in operator and define loggers in packages instead injecting them here?: https://github.com/hazelcast/hazelcast-platform-operator/blob/main/api/v1alpha1/hazelcast_webhook.go#L15

This has also additional benefit of avoiding the need to pass logger in most of the function calls. Cleaner API.

**Review by dzeromski-hazelcast — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (kutluhanmetin) [reply go.mod]: Would you please check it again?

File: main.go (hunk 10 above)
Comment (kutluhanmetin) [reply main.go]: Would you please check it again?

**Review by hasancelik — APPROVED** [review]

**Pull request merged**